Diary/ListScene/Controller/DiaryViewController.swift

```diff
@@ -0,0 +1,192 @@
+        setUpRightItem()
+    }
+    
+    @objc private func moveRegisterViewController() {
+        let viewContoller = UpdateViewController()
+        
+        navigationController?.pushViewController(
+            viewContoller,
+            animated: true
+        )
+    }
+    
+    private func setUpTableView() {
+        tableView.register(DiaryCell.self)
+        tableView.register(UITableViewCell.self)
+        tableView.delegate = self
+    }
+    
+    private func setUpTableViewLayout() {
+        view.addSubview(tableView)
+        
+        tableView.translatesAutoresizingMaskIntoConstraints = false
+        
+        NSLayoutConstraint.activate([
+            tableView.topAnchor.constraint(equalTo: view.topAnchor),
+            tableView.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+            tableView.bottomAnchor.constraint(equalTo: view.bottomAnchor),
+            tableView.trailingAnchor.constraint(equalTo: view.trailingAnchor)
+        ])
+    }
+    
+    private func setUpDataSource() {
+        dataSource = UITableViewDiffableDataSource<Int, DiaryDTO>(tableView: tableView) {
+            tableView, indexPath, itemIdentifier in
+            
+            if let cell = tableView.dequeueReusableCell(
+                withIdentifier: DiaryCell.identifier,
+                for: indexPath
+            ) as? DiaryCell {
+                cell.configure(data: itemIdentifier)
+                
+                return cell
+            }
+            
+            return tableView.dequeueReusableCell(
+                withIdentifier: UITableViewCell.identifier,
+                for: indexPath
+            )
+        }
+    }
+    
+    private func setUpSampleData() {
+        guard let sampleData: [DiaryDTO] = AssetManager.get() else {
+            return
+        }
+        setUpSanpshot(data: sampleData)
+    }
+    
+    private func setUpCoreData() {
+        guard let coreData = DiaryDAO.shared.read() else {
+            return
+        }
+        
+        setUpSanpshot(data: coreData)
+    }
+    
+    private func setUpSanpshot(data: [DiaryDTO]) {
+        var snapshot = NSDiffableDataSourceSnapshot<Int, DiaryDTO>()
+        
+        snapshot.appendSections([.zero])
+        snapshot.appendItems(data)
+        
+        dataSource?.apply(snapshot)
+    }
+}
+
+extension DiaryViewController: UITableViewDelegate {
+    func tableView(
+        _ tableView: UITableView,
+        didSelectRowAt indexPath: IndexPath
+    ) {
+        guard let cell = tableView.cellForRow(at: indexPath) as? DiaryCell,
+              let diaryData = dataSource?.itemIdentifier(for: indexPath) else {
+            return
+        }
+        
+        let viewContoller = UpdateViewController(diaryData: diaryData)
+        
+        cell.isSelected = false
+        
+        navigationController?.pushViewController(
+            viewContoller,
+            animated: true
+        )
+    }
```

Comment:
UpdateViewController의 프로퍼티 diaryData가 nil혹은 값이있는지 여부로 수정페이지 인지 추가 페이지인지 구분하고 있습니다!

그래서 @objc private func moveRegisterViewController() 메서드를 사용해 수정페이지나 추가페이지로 push 하기위해서는 diaryData를 매개변수로 받아야할것같은데 @objc 메서드의 경우 매개변수로 받을수 있는것이 Gesture, Sender 만 받을수 있는것 같습니다! 아래와 같은 에러가 납니다!

`Method cannot be marked @objc because the type of the parameter cannot be represented in Objective-C`

이를 해결할 좋은 방법이 있을까요??🤔  조언을 받고싶습니다!🫣